plugin.js

```diff
@@ -682,189 +496,423 @@
 		};
 	}
 
-	placeholdersCollection = new PlaceholdersCollection();
+	/**
+	 * @class
+	 * @constructor
+	 * @param {CKEDITOR.editor} editor
+	 */
+	function PlaceholderElementsPlugin(editor)
+	{
+		var _this = this;
 
-	CKEDITOR.plugins.add('placeholder_elements', {
-		hidpi: true,
-		icons: 'placeholderelements',
-		lang: ['en', 'de'],
-		requires: ['richcombo', 'widget'],
+		var config = editor.config.placeholder_elements;
+		var placeholdersCollection = new PlaceholdersCollection();
+		var placeholdersRegexLoose;
+		var placeholdersRegexExact;
+		var placeholdersMap = {};
+		var uiElement;
+		var menuButtonStates = {};
+		var menuButtonGroup = 'placeholderElementsButton';
+		var contentIsUpdating = false;
+		var contentNeedsUpdating = false;
+		var selectionChangeProcessor;
 
-		placeholders: placeholdersCollection,
+		/**
+		 * @type {PlaceholdersCollection}
+		 */
+		_this.placeholders = placeholdersCollection;
 
-		init: function(ed)
+		/**
+		 * Processes text and replaces tokens with the appropriate html elements.
+		 *
+		 * @param text
+		 * @returns {XML|void|string}
+		 */
+		_this.dataFilterTextRule = function(text)
 		{
-			editor = ed;
+			return text.replace(placeholdersRegexLoose, function(match)
+			{
+				var element = new CKEDITOR.htmlParser.element('span', {
+					'class': 'cke_placeholder_element'
+				});
+				element.add(new CKEDITOR.htmlParser.text(match));
 
-			var defaults = {
-				css:
-					'.cke_placeholder_element { background: #ffff00; } ' +
-					'a .cke_placeholder_element { text-decoration: underline }',
-				draggable: false,
-				placeholders: [],
-				startDelimiter: '{',
-				endDelimiter: '}',
-				uiType: 'button'
-			};
-			config =
-				editor.config.placeholder_elements =
-					CKEDITOR.tools.extend(editor.config.placeholder_elements || {}, defaults);
+				return editor.widgets.wrapElement(element, 'placeholder_elements').getOuterHtml();
+			});
+		};
 
-			CKEDITOR.addCss(config.css);
+		/**
+		 * Cleans up all listeners related to this instance of the plugin.
+		 */
+		_this.destroy = function()
+		{
+			_this.placeholders.removeAllListeners();
+			if(selectionChangeProcessor !== undefined)
+			{
+				editor.removeListener('selectionChange', selectionChangeProcessor);
```

Comment:
This shouldn't be necessary, as the editor removes all listeners before `instanceDestroyed` is being fired. On a side note, I don't think it would work that way anyways, as `on()` doesn't return the listener function, but a new object.